core/src/main/java/io/questdb/griffin/engine/functions/str/SplitPartVarcharFunctionFactory.java

```diff
@@ -0,0 +1,224 @@
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2024 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.griffin.engine.functions.str;
+
+import io.questdb.cairo.CairoConfiguration;
+import io.questdb.cairo.sql.Function;
+import io.questdb.cairo.sql.Record;
+import io.questdb.cairo.sql.SymbolTableSource;
+import io.questdb.griffin.FunctionFactory;
+import io.questdb.griffin.SqlException;
+import io.questdb.griffin.SqlExecutionContext;
+import io.questdb.griffin.engine.functions.TernaryFunction;
+import io.questdb.griffin.engine.functions.VarcharFunction;
+import io.questdb.griffin.engine.functions.constants.VarcharConstant;
+import io.questdb.std.IntList;
+import io.questdb.std.Mutable;
+import io.questdb.std.Numbers;
+import io.questdb.std.ObjList;
+import io.questdb.std.str.Utf8Sequence;
+import io.questdb.std.str.Utf8Sink;
+import io.questdb.std.str.Utf8StringSink;
+import io.questdb.std.str.Utf8s;
+import org.jetbrains.annotations.Nullable;
+
+
+public class SplitPartVarcharFunctionFactory implements FunctionFactory {
+    @Override
+    public String getSignature() {
+        return "split_part(ØØI)";
+    }
+
+    @Override
+    public Function newInstance(int position, ObjList<Function> args, IntList argPositions, CairoConfiguration configuration, SqlExecutionContext sqlExecutionContext) throws SqlException {
+        final Function varcharFunc = args.getQuick(0);
+        final Function delimiterFunc = args.getQuick(1);
+        final Function indexFunc = args.getQuick(2);
+        final int indexPosition = argPositions.getQuick(2);
+
+        if (indexFunc.isConstant()) {
+            int index = indexFunc.getInt(null);
+            if (index == Numbers.INT_NULL) {
+                return VarcharConstant.NULL;
+            } else if (index == 0) {
+                throw SqlException.$(indexPosition, "field position must not be zero");
+            } else {
+                return new SplitPartVarcharConstIndexFunction(varcharFunc, delimiterFunc, indexFunc, indexPosition, index);
+            }
+        } else if (!indexFunc.isRuntimeConstant()) {
+            throw SqlException.$(indexPosition, "index must be a constant or runtime-constant");
```

Comment:
Since this message is what the user sees, I think the term "runtime-constant" won't be immediately clear. Suggestion:

`index must be either a constant expression or a placeholder`